R/convertToEGRET.R

```diff
@@ -0,0 +1,75 @@
+#' Convert loadflex to EGRET object
+#'  
+#' Convert a loadflex object into an EGRET object for plotting.
+#' 
+#' @param intdat data.frame of interpolation data
+#' @param estdat data.frame of estimation data
+#' @param preds data.frame of load predictions
+#' @param meta loadflex metadata object; it must include constituent,
+#' flow, conc.units, custom (station abbreviation: staAbbr)
+#' 
+#' @importFrom dplyr left_join
+#' @importFrom dplyr rename_
+#' @importFrom dplyr mutate
+#' @importFrom lubridate decimal_date
+#' @importFrom EGRET as.egret
+#' 
+#' @export
+convertToEGRET <- function(intdat, estdat, preds, meta){
+  
+  # flow_col <- verify_meta(meta, 'flow')
+  flow_col <- "DISCHARGE"
+  info_df <- data.frame(shortName=verify_meta(meta, 'station'),
+                        paramShortName='nitrate',
+                        staAbbrev=verify_meta(meta, c('custom', 'staAbbr')),
+                        # constitAbbrev=verify_meta(meta, 'constituent'),
+                        constitAbbrev='NO3',
+                        # param.units=verify_meta(meta, 'conc.units'),
+                        param.units='mg L^-1',
+                        stringsAsFactors = F)
+  
+  discharge_df <- estdat %>% 
+    left_join(preds, by=c("DATE" = "date")) %>% 
+    rename_(.dots = setNames(c("DATE", flow_col, "fit"),
+                            c("Date", "Q", "ConcDay"))) %>% 
+    mutate(LogQ=log(Q))
+  
+  sample_df <- intdat %>% 
+    rename_(.dots=setNames(c("DATE", info_df$constitAbbrev, flow_col), 
+                           c("Date", "ConcLow", "Q")))  %>% 
+    mutate(ConcHigh=ConcLow, 
+           ConcAve=mean(c(ConcLow, ConcHigh)), 
+           Uncen=rep(1, nrow(intdat)), 
+           LogQ=log(Q),
+           DecYear=lubridate::decimal_date(Date))
+  
+  eList <- as.egret(INFO=info_df, Daily=discharge_df, Sample=sample_df)
+  return(eList)
+}
+
+
+#' Verify metadata object
+#'  
+#' Verify that the loadflex metadata object has everything it needs.
+#' 
+#' @param meta loadflex metadata object
+#' @param nm character name of the metadata item to check. If it is a 
+#' custom name, this would be a character vector with the first name as
+#' 'custom' (e.g. nm = c('custom', 'staAbbr'))
+#' 
+#' @export
+verify_meta <- function(meta, nm){
```

Comment:
yeah, good thought - let's keep it internal at least for now. someday i'll look into the validity thing; i think it's reasonable to keep the functions separate for the time being.